zulipterminal/model.py

```diff
@@ -1206,6 +1206,20 @@ def _handle_update_message_event(self, event: Event) -> None:
                     else:
                         self.index["topics"][stream_id] = []
 
+            # If the user is composing a message to this topic, renarrow by
+            # updating the topic in the title_write_box header. The
+            # renarrowing is justified if enough of the topic has been moved
+            # i.e., a 'change_later' or 'change_all' change in the topic.
+            # The logic here has been inspired by the web app's
+            # `static/js/message_events.js`.
+            if event["propagate_mode"] in ["change_later", "change_all"]:
+                if hasattr(self.controller, "view"):
+                    view = self.controller.view
+                    old_subject = event["orig_subject"]
+                    if view.write_box.stream_id == stream_id:
+                        if view.write_box.title_write_box.edit_text == old_subject:
+                            view.write_box.update_topic_compose_header(new_subject)
```

Comment:
Do you mean extending the "alias" further to `self.controller.view.write_box` and giving that a name instead? So that it makes the code below more compact.

Reply:
That's what I meant by the last sentence, yes.